apis/fluentbit/v1alpha2/collector_types.go

```diff
@@ -0,0 +1,132 @@
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/fluent/fluent-operator/pkg/utils"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// CollectorSpec defines the desired state of FluentBit
+type CollectorSpec struct {
+	// Fluent Bit image.
+	Image string `json:"image,omitempty"`
+	// Fluent Bit Watcher command line arguments.
+	Args []string `json:"args,omitempty"`
+	// Fluent Bit image pull policy.
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
+	// Fluent Bit image pull secret
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+	// Compute Resources required by container.
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+	// NodeSelector
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// Pod's scheduling constraints.
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+	// Tolerations
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// Fluentbitconfig object associated with this Fluentbit
+	FluentBitConfigName string `json:"fluentBitConfigName,omitempty"`
+	// The Secrets are mounted into /fluent-bit/secrets/<secret-name>.
+	Secrets []string `json:"secrets,omitempty"`
+	// RuntimeClassName represents the container runtime configuration.
+	RuntimeClassName string `json:"runtimeClassName,omitempty"`
+	// PriorityClassName represents the pod's priority class.
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+	// List of volumes that can be mounted by containers belonging to the pod.
+	Volumes []corev1.Volume `json:"volumes,omitempty"`
+	// Pod volumes to mount into the container's filesystem.
+	VolumesMounts []corev1.VolumeMount `json:"volumesMounts,omitempty"`
+	// Annotations to add to each Fluentbit pod.
+	Annotations map[string]string `json:"annotations,omitempty"`
+	// SecurityContext holds pod-level security attributes and common container settings.
+	SecurityContext *corev1.PodSecurityContext `json:"securityContext,omitempty"`
+	// Host networking is requested for this pod. Use the host's network namespace. If this option is set, the ports that will be used must be specified. Default to false.
+	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// PVC definition
+	PersistentVolumeClaim *corev1.PersistentVolumeClaim `json:"pvc,omitempty"`
+	// RBACRules represents additional rbac rules which will be applied to the fluent-bit clusterrole.
+	RBACRules []rbacv1.PolicyRule `json:"rbacRules,omitempty"`
+	// By default will build the related service according to the globalinputs definition.
+	DisableService bool `json:"disableService,omitempty"`
+	// The path where buffer chunks are stored. This field would make no effect in memory buffer plugin.
+	// +kubebuilder:validation:Required
+	BufferPath *string `json:"bufferPath,omitempty"`
```

Comment:
`kubebuilder:validation:Required` and `omitempty` should not be set simultaneously.